client/src/lib.rs

```diff
@@ -309,6 +322,34 @@ impl<C> ConfiguredClient<C> {
     }
 }
 
+#[derive(Debug)]
+struct ClientHeaders {
+    user_headers: HashMap<String, String>,
+    api_key: Option<String>,
+}
+
+impl ClientHeaders {
+    fn apply_to_metadata(&self, metadata: &mut MetadataMap) {
```

Comment:
I considered doing some of this work on set rather than apply (md key/val creation, concat w/ "Bearer "), but it didn't seem too burdensome to do it on the call

Reply:
This makes more sense to me anyway I think. Sure, it's "extra" work but totally irrelevant to performance and this is a bit more flexible